### PR TITLE
Render shape label while drawing

### LINKED ISSUE #6540: Render `newshape` `text` and `texttemplate` during initial shape draw
Currently, if a shape label is defined in `newshape` using `text` or `texttemplate`, the text is not rendered until mouseup following the initial draw.

Modify shape drawing code such that the text is rendered during mousedown.

This is doable, but involves some additional complexity since the initial shape drawing code (during mousedown) is completely separate from the normal shape drawing code, and we don't have access to the same set of variables.

Implementation will involve modifying key lines in `src/components/selections/select.js` and possibly `src/components/shapes/display_outlines.js` due to the overlap between selection and shape code.

### PR DESCRIPTION
**Pull request by emilykl**
Closes #6540 

Render shape label text during initial shape draw.

Codepen demo here: https://codepen.io/emily_plotly/pen/eYLrKNV

This PR leverages the existing newshape code to render the shape label while the shape is being drawn.

@alexcjohnson @archmoj I'm very interested to get your feedback on whether this approach makes sense and whether there might be any unintended side effects.

Note: I moved the `drawLabels()` function from `shapes/draw.js` into its own file; that makes up the bulk of the changes in terms of # of lines. No changes to that function itself.

Outstanding changes:

- [ ] Fix broken Jasmine tests
- [x] Make label partially transparent like the shape itself

### CODE REVIEW

File: src/components/shapes/display_labels.js [hunk 1]
```diff
@@ -0,0 +1,268 @@
+'use strict';
```
Comment (emilykl) [inline src/components/shapes/display_labels.js:1]: This is just the `drawLabels()` function which used to be in `shapes/draw.js` being moved into its own file. No code changes.

File: src/components/shapes/draw_newshape/newshapes.js [hunk 2]
```diff
@@ -70,8 +62,65 @@ module.exports = function newShapes(outlines, dragOptions) {
         }
     }
 
-    var isOpenMode = openMode(dragmode);
+    var newShape = createShapeObj(outlines, dragOptions);
```
Comment (emilykl) [inline src/components/shapes/draw_newshape/newshapes.js:65]: The only actual change here is moving some of the logic into a separate `createShapeObj()` function

**Comment by archmoj** [comment]
Nicely done.
/ :dancer: /
\ :dancer: \

**Comment by emilykl** [comment]
Tagging @JulianWgs as well!

**Pull request merged**